app/src/main/java/com/navapp/navigation/util/geofencing/UponArrival.java

```diff
@@ -0,0 +1,12 @@
+package com.navapp.navigation.util.geofencing;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class UponArrival extends BroadcastReceiver {
```

Comment:
Pretty sure it needs to be defined in the manifest

Reply:
Only if like global for UI or something, I don't think that in this use case I need to do so.

Reply:
"If you declare a broadcast receiver in your manifest, the system launches your app (if the app is not already running) when the broadcast is sent"

Reply:
Then check